service/app.js

```diff
@@ -0,0 +1,282 @@
+
+app.post('/createCards', async (req, res) => {
+  const cardRequest = req.body
+  console.log(cardRequest)
+  if ((!cardRequest.frontImage && !cardRequest.frontText) || 
+    (!cardRequest.backImage && !cardRequest.backText)) {
+    res.status(400).send('Card data incomplete')
+  }
+
+  if ((frontImage && !isUrl(frontImage)) || (backImage && !isUrl(backImage))) {
+    res.status(400).send('Image fields must be valid URLs')
+  }
+
+  if (!cardRequest.deckId) {
+    res.status(400).send('Deck ID is required')
+  }
+
+  try {
+    const deck = await Deck.findById(cardRequest.deckId)
+    
+    if (deck) {
+      deck.createCards.push({
+        frontImage: cardRequest.frontImage,
+        frontText: cardRequest.frontText,
+        backImage: cardRequest.backImage,
+        backText: cardRequest.backText
+      })
+       deck.save()
+      
+      res.sendStatus(204)
+    } else {
+      res.sendStatus(404)
+    }
+  } catch (err) {
+    console.log(`error in creating card ${err}`)
+    res.sendStatus(502)
+  }
+})
+
+//===============================================create deck===========================================================================
+app.post('/createDeck', async (req, res) => {
+  const userRequest = req.body
+  console.log(userRequest)
+  if(deck){
+    
+  }
+    try {
+      const deck = await Deck.create(userRequest)
+      const result = await deck.save()
+      return res.sendStatus(200)
+    } catch (error) {
+      console.log("error creating decks")
+    }
+
+})
+
+//===============================================create user===========================================================================
+//i beliieve this wont be pushed since we have not created the user on mongodb
+app.post("/createUser", async (req, res) => {
+  const user = await userInfo.create(req.params.createUsers)
+  try {
+    const result = await user
+    if (result){
+      result.createUsers.push({
+          firstName:user.firstName,
+          lastName :user.lastName,
+          userId:   user.userId
+      })
+      result.save()
+      console.log(user)
+     return res.status(201)
+    }else{
+      res.sendStatus(404)
+    }
+  } catch (error) {
+    console.log('user not created')
+  }
+})
+
+//==========================================Delete user============================================================================
+
+app.post('"/deleteUser/:id"', async (req, res) => {
+  const user = await userInfo.findById(req.params.id)
+  try {
+    if(user) {
+      await user.remove()
+      res.sendStatus(200)
+    } else {
+       res.status(404)
+    }
+  } catch (error) {
+     res.status(404).json({
+    error: "user does not exists"
+    })
+  }
+})
+
+//=======================================update card=================================================================================
+
+app.put('/updatecard/:Id', async (req, res) => {
+  const cardRequest = req.body
+
+  try {
+    const deck = await Deck.findById(cardRequest.deckId)
```

Comment:
Passing the deck id in the request is one way to accomplish this, but it is a departure from the other routes we've designed. Think about how you might pass the deck id in the path.